src/main/app/controller/GetController.ts

```diff
@@ -40,7 +38,27 @@ export class GetController {
       ...commonPageContent,
       ...commonLanguageContent,
       sessionErrors,
-      ...(this.stepId && { formState: req.session.state[this.stepId] }),
+      ...(this.stepId && { formState: req.session.state?.[this.stepId] }),
     });
   }
+
+  private getContent(req: AppRequest, res: Response, translations: Translations): Translations {
+    if (typeof this.content !== 'function') {
+      return this.content;
+    }
+
+    const isDivorce = res.locals.serviceType !== 'civil';
+    const yourDetails = req.session.state?.['your-details'] as Record<string, string> | undefined;
+    const selectedPartnerGender = yourDetails?.partnerGender;
+    let partner = '';
```

Comment:
rather than  a let maybe you could split out getPartner into a method with return values

Reply:
👍 Good idea, I've added this in cbd0408.